core/templates/pages/contributor-dashboard-page/translation-language-selector/translation-language-selector.component.spec.ts

```diff
@@ -258,4 +262,33 @@ describe('Translation language selector', () => {
         selectedLanguage);
     });
   });
+
+  it('should toggle dropdown', () => {
```

Comment:
Does this test both the cases for `filterDivRef`, .i.e, when `if (this.filterDivRef) {` returns `True` and `False`?

Reply:
It only tests the True value. We never expect `this.filterDivRef` to be null inside the `setTimeout` callback as it has 1 ms in the second parameter. However, without the `if` check, tests are flaky as `this.filterDivRef` sometimes becomes undefined in the test environment, so I added the the `if` check to prevent flakiness.

Reply:
have you tried using `tick()` to account for the wait time?

Reply:
Done, using flush instead of tick().